### PR TITLE
cross subjet and two paradigm

### PR DESCRIPTION
**Pull request by chris-mrn**
we have now two different paradigm and we can choose to train and test on different subjects.

### CODE REVIEW

**Review by tomMoral — COMMENTED** [review]
A few comments.

File: objective.py [hunk 1]
```diff
@@ -15,34 +16,93 @@ class Objective(BaseObjective):
...
+
+            X_0 = session_X[self.cross_session[0]-1]
+            y_0 = session_y[self.cross_session[0]-1]
+            X_1 = session_X[self.cross_session[1]-1]
+            y_1 = session_y[self.cross_session[1]-1]
+
+            X_train_0, X_test_0, y_train_0, y_test_0 = train_test_split(X_0,
+                                                                        y_0)
+            X_train_1, X_test_1, y_train_1, y_test_1 = train_test_split(X_1,
+                                                                        y_1)
```
Comment (tomMoral) [inline objective.py:100]: Sam here, train on all session except one and test on the last one.

File: solvers/CSPlda.py [hunk 2]
```diff
@@ -20,6 +20,9 @@ class Solver(BaseSolver):
 
     stopping_criterion = SingleRunCriterion()
 
+    requirements = ['mne']
+    install_cmd = 'conda'
```
Comment (tomMoral) [inline solvers/CSPlda.py:24]: Exchange

**Review by tomMoral — COMMENTED** [review]
Some more comments

File: datasets/BNCI_M.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    def get_data(self):
+
+        # The return arguments of this function are passed as keyword arguments
+        # to `Objective.set_data`. This defines the benchmark's
+        # API to pass data. It is customizable for each benchmark.
+        dataset = BNCI2014001()
+        paradigm = MotorImagery(n_classes=3)
+        # The dictionary defines the keyword arguments for `Objective.set_data`
+        return dict(dataset=dataset, paradigm=paradigm)
```
[the remark this answers is not in the document]
Comment (chris-mrn) [reply datasets/BNCI_M.py:31]: Yes I agree 👍

File: objective.py [hunk 4]
```diff
@@ -15,34 +16,93 @@ class Objective(BaseObjective):
...
+            X_0, y_0, _ = paradigm.get_data(dataset=dataset,
+                                            subjects=[self.cross_subject[0]])
+
+            X_1, y_1, _ = paradigm.get_data(dataset=dataset,
+                                            subjects=[self.cross_subject[1]])
+
+            X_train_0, X_test_0, y_train_0, y_test_0 = train_test_split(X_0,
+                                                                        y_0)
+            X_train_1, X_test_1, y_train_1, y_test_1 = train_test_split(X_1,
+                                                                        y_1)
```
[the remark this answers is not in the document]
Comment (chris-mrn) [reply objective.py:72]: I did this in order to evaluate the objective value of intra_subject and inter_subject using the same amount of training data.

[the remark this answers is not in the document]
Comment (chris-mrn) [reply objective.py:72]: Okay i will look at it 👍

File: objective.py [hunk 5]
```diff
@@ -15,34 +17,90 @@ class Objective(BaseObjective):
...
+            X_test = session_X[n_session-1]
+            X_train = np.array(flatten(session_X[:(n_session-1)]))
+            y_test = session_y[n_session-1]
+            y_train = np.array(flatten(session_y[:(n_session-1)]))
+
+            self.X_train, self.y_train = X_train, y_train
+            self.X_test, self.y_test = X_test, y_test
+            return dict(X_train=X_train, y_train=y_train,
+                        X_test=X_test, y_test=y_test
+                        )
```
Comment (tomMoral) [inline objective.py:84]:
```suggestion
            return dict(
                X_train=X_train, y_train=y_train,
                X_test=X_test, y_test=y_test
            )
```

**Review by tomMoral — COMMENTED** [review]
two extra nitpicks but this looks nice!

**Comment by tomMoral** [comment]
can you share the result of a small run?

**Comment by tomMoral** [comment]
Is this PR still relevant after the big changes in #3?

**Comment by chris-mrn** [comment]
No it isn't relevant anymore

**Pull request closed without merging**